### PR TITLE
#27500 Prepare OfflineShipping module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7398](https://jira.corp.magento.com/browse/ENGCOM-7398) has been created to process this Pull Request

File: app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php [hunk 1]
```diff
@@ -12,7 +12,7 @@ class ColumnResolver
     const COLUMN_REGION = 'Region/State';
     const COLUMN_ZIP = 'Zip/Postal Code';
     const COLUMN_WEIGHT = 'Weight (and above)';
-    const COLUMN_WEIGHT_DESTINATION = 'Weight (and above)';
```
Comment (slavvka) [inline app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: Why did you change resource model?

Comment (lbajsarowicz) [reply app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: PHPStan reported that values are overriden on lines 25 - 26. I verified implementation and noticed that author of the test actually used coincidence to make his tests pass. :-)

Comment (slavvka) [reply app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: How can affect the functionality? Should it be covered by test?

Comment (lbajsarowicz) [reply app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: We can just remove the test that refers to these `const`s :-) or create `private const` inside the test with correct consts, and leave resource model still wrong :-)

Comment (slavvka) [reply app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: @lbajsarowicz In order to deliver updated tests as easy as possible our intention is not to change anything except for tests. That will allow us to run only Unit and Static tests and avoid additional load for regression. You can create a separate PR that fixes the issue

Comment (lbajsarowicz) [reply app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: Ok. I'll put the phpstan ignore on that error then and rollback  fix

Comment (slavvka) [reply app/code/Magento/OfflineShipping/Model/ResourceModel/Carrier/Tablerate/CSV/ColumnResolver.php]: But please create a separate fix just not to lose it :)

**Review by slavvka — CHANGES_REQUESTED** [review]
Please revert the change of the model and create another PR to address the issue

**Comment by lbajsarowicz** [comment]
Ticket created.

**Review by slavvka — CHANGES_REQUESTED** [review]
@lbajsarowicz please fix static tests

**Comment by lbajsarowicz** [comment]
Rebased & forced @slavvka

**Comment by slavvka** [comment]
```
14:56:48 There were 13 warnings:
14:56:48 
14:56:48 1) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_and_new_status" (null, null, null)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 2) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_status_and_subscribe" (null, true, true)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 3) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "new_unsubscribed_value_and_missing_previous_status" (null, false, null)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 4) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_subscribed_status_without_new_value" (1, null, null)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 5) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_subscribed_previous_and_new_status" (1, true, null)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 6) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_subscribed_customer" (1, false, false)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 7) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previously_unsubscribed_status_without_new_value" (3, null, null)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 8) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unsubscribed_customer" (3, true, true)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 9) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_unsubscribed_previous_and_new_status" (3, false, null)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 10) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_unconfirmed_status_without_new_value" (4, null, true)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 11) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unconfirmed_status" (4, true, true)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 12) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_unconfirmed_status" (4, false, true)
14:56:48 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
14:56:48 
14:56:48 13) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterGetByIdCreatesExtensionAttributes
14:56:48 Trying to configure method "setIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
```
